### PR TITLE
staterror support

### PR DESCRIPTION
**Pull request by lukasheinrich**
# Description

Starting to work on #85 .

ROOT implementation links: 

https://root.cern.ch/doc/v610/HistoToWorkspaceFactoryFast_8cxx_source.html#l02424

this looks more convoluted in ROOT than it is (at least for basic support), but keeping this as reference

Todo:

- [x] add sufficient information (just need nominal counts and uncertainties for each sample)
- [x] calculate overall uncertainty from individual sample uncertainties (~addition in quadrature)
- [x] implement constraint term (default seems gaussian, using this for now)

# Checklist Before Requesting Approver
- [ ] Tests are passing
- [ ] "WIP" removed from the title of the pull request

### CODE REVIEW

**Comment by kratsg** [comment]
Don't forget to update `modifiers/__init__.py` towards the bottom to make sure it's auto-loaded/auto-registered (by default it won't be).

**Comment by lukasheinrich** [comment]
the implementation seems ok.. just need to add some tests.. comparigon with ROOT (we really need #3, this overlay business is silly :-p )

screenshot

**Comment by lukasheinrich** [comment]
@kratsg I added a comment and also a test to make sure the computation is right. Maybe you can take a look, the tests should pass and from my side, i'm happy w/ th epR

File: pyhf/modifiers/staterror.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    def alphas(self, pars):
+        return pars  # nuisance parameters are also the means of the
+
+    def pdf(self, a, alpha):
+        tensorlib, _ = get_backend()
+        # this computes sum_i uncertainty_i for all samples
+        # (broadcastted for all bins in the channel)
+        # for each bin, the relative uncert is the width of a gaussian
+        # which is the constraint pdf; Prod_i Gaus(x = 1, mu = alpha_i, sigma = relunc_i)
```
Comment (kratsg) [inline pyhf/modifiers/staterror.py]: Shouldn't this be `x=a`?

Comment (lukasheinrich) [reply pyhf/modifiers/staterror.py]: done

**Comment by kratsg** [comment]
Remove WIP and LGTM.

**Pull request merged**